mypy/typeops.py

```diff
@@ -710,7 +710,9 @@ def callable_type(
     fdef: FuncItem, fallback: Instance, ret_type: Type | None = None
 ) -> CallableType:
     # TODO: somewhat unfortunate duplication with prepare_method_signature in semanal
-    if fdef.info and not fdef.is_static and fdef.arg_names:
+    if fdef.name == "__new__":
```

Comment:
Hmm, mutating fdef here seems a little suspicious to me. Why do we need this? (My weak model for when it's safe to mutate these kind of things is only during semantic analysis, like you do in prepare method signature)

Reply:
Ah, good catch, thanks! I'm actually not entirely sure why I added this -- perhaps this was before I made sure the semantic analyzer set the is_static property. At any rate, I've verified that is_static is set correctly for methods named `__new__` by the time we get in typeops, meaning that it is not necessary to modify the attribute here. I've removed it.

Reply:
Thanks! Github is having some problems, so your commit hasn't shown up. The workaround I found is to make an empty commit, push it, then reset to delete the commit, then force push

Reply:
I actually had some issues with my local git client suddenly deciding to use the wrong SSH key -- fixed that now, but I see that the commit still isn't visible. Will make an empty commit and push again, thanks for the tip!